app/src/test/java/waiter/HttProtocolTest.java

```diff
@@ -57,15 +67,50 @@ void returnsOkWithNoBodyWhenHeadMethodWithExistingUrl() {
         assertTrue(response.endsWith("\r\n\r\n"));
     }
 
+    @Test
+    void returnsOkWithBodyWhenPostMethodWithExistingUrl() {
+        String url = "/echo_body";
+        String body = "foo";
+        String request = String.format(
+                """
+                POST %s HTTP/1.1
+                Content-Length: %d
+                
+                %s""", url, body.length(), body).replace("\n", "\r\n");
```

Comment:
Non-blocking: Not asking you to change this, but FWIW I find this somewhat harder to read than just the plain text of the request, i.e. no separate url, body, call to `replace`, etc.

Reply:
+1